Switch to Rails

gabebw opened the pull request:
I visited the homepage and every link in the header (About, Donate, Resources).
They are all exactly the same as the design on railsbridgeboston.org (I
switched back and forth between tabs to double check).

gabebw commented:
Uhhhh what if I don't fix any of these problems in autogenerated files?

gabebw commented on spec/lib/eventbrite_spec.rb:
```diff
@@ -25,19 +25,20 @@
-      response = { events: [{ "title" => "event info" }] }
+      events = { events: [{ "title" => "event info" }] }
 
-      stub_request(:get, eventbrite_api_url).to_return(body: { id: "123" })
```
`stub_request` with the same method and URL overwrites the stub each time, so this stub was never working

gabebw commented on spec/lib/eventbrite_spec.rb:
```diff
@@ -25,19 +25,20 @@
       end
 
       def stub_failed_eventbrite_request
-        stub_request(:get, eventbrite_api_url).to_return(body: { id: "123" })
```
This stub was overwritten by the next one, so I folded it into the next one

mxie replied:
Oops. That totally makes sense. What was I thinking?

mxie commented on Gemfile:
```diff
@@ -1,24 +1,61 @@
 gem "nokogiri"
-gem "httparty"
+gem "normalize-rails", "~> 3.0.0"
+gem "pg"
```
We're not actually using a database, can we omit this?

mxie commented on Gemfile:
```diff
@@ -1,24 +1,61 @@
+gem "pg"
+gem "rack-contrib"
+gem "rails", "4.2.1"
+gem "recipient_interceptor"
```
I don't think we have to worry about emails either.

mxie commented on README.md:
```diff
@@ -1,33 +1,32 @@
-RailsBridge Boston Website
-==========================
+# Railsbridgeboston dot org
```
Let's revert this header back to the original.

mxie commented on README.md:
```diff
@@ -1,33 +1,32 @@
-==========================
+# Railsbridgeboston dot org
 
-[![Build
```
Let's keep this in!

gabebw replied:
oooooops

mxie commented on README.md:
```diff
@@ -1,33 +1,32 @@
-Status](https://travis-ci.org/railsbridge-boston/railsbridgeboston_dot_org.svg?branch=master)](https://travis-ci.org/railsbridge-boston/railsbridgeboston_dot_org)
+## Getting Started
 
-This is the website for RailsBridge Boston, a chapter of [RailsBridge]. You can
```
We should keep these informational paragraphs.

gabebw commented on Gemfile:
```diff
@@ -1,24 +1,54 @@
-gem "middleman-livereload"
-gem "middleman-dotenv"
-gem "rack-contrib"
-gem "compass", "~> 1.0.0.alpha.20"
```
We're not actually using Compass.

mxie commented on app/assets/stylesheets/_resources.scss:
```diff
@@ -1,4 +1,4 @@
-.resources_index {
+.resources {
```
Wait, was this intentional?

gabebw replied:
Yes, it fixes the design and has no side effects that I could see.

Middleman gives one of the pages a class of `resources resources_index`, so I gave the appropriate page in Rails a class of `resources`. I could have used `resources_index` too, I suppose.

mxie replied:
Either way is fine, as long as we're consistent and not breaking all the pretty styles Connie has put into place!

gabebw replied:
They are pixel-perfectly exact.

mxie commented on app/views/pages/about.html.erb:
```diff
@@ -37,10 +37,10 @@ title: About - RailsBridge Boston
-        <li class="contact-<%= contact.type %>">
-          <%= contact.description %>
-          <a href="<%= contact.url %>"><%= contact.title %></a>
+    <% data_links["contacts"].each do |contact| %>
```
Indent 1 more level in.

mxie commented on app/views/pages/about.html.erb:
```diff
@@ -49,10 +49,10 @@ title: About - RailsBridge Boston
         <li>
-          <a href="<%= friend.url %>"><%= friend.name %></a>:
-          <%= friend.description %>
+          <a href="<%= friend["url"] %>"><%= friend["name"] %></a>:
```
How about `link_to` here instead?

mxie commented on app/views/pages/about.html.erb:
```diff
@@ -37,10 +37,10 @@ title: About - RailsBridge Boston
+    <% data_links["contacts"].each do |contact| %>
+        <li class="contact-<%= contact["type"] %>">
+          <%= contact["description"] %>
+          <a href="<%= contact["url"] %>"><%= contact["title"] %></a>
```
`link_to`?

mxie commented:
This looks pretty good Gabe! The only other thing is that I'd like to remove anything that involves the database. I can't foresee a situation where we might need it, so I don't think it makes sense to keep that stuff around. Thoughts?

gabebw commented:
I think I removed everything around the DB - I searched for `activerecord` and `active_record` and removed those settings, and also disabled the AR Railtie. I removed DelayedJob and PG from the Gemfile and removed all of their configuration too.

gabebw commented:
Was that a provisional :+1:, Melissa?

gabebw commented:
@mxie

gabebw commented:
OK, I'm merging.
